src/api/agendaItem.ts

```diff
@@ -61,23 +61,70 @@ export interface AgendaItem {
   neighbourhoodId?: number[];
 }
 
-export const fetchItems = async () => {
-  const now = new Date();
-  const nextMonth = new Date();
-  nextMonth.setMonth(nextMonth.getMonth() + 1);
+type AgendaItemFetchOptions = {
+  start: Date;
+  end: Date;
+};
+
+const fetchItemPage = async (
+  page: number,
+  { start, end }: AgendaItemFetchOptions,
+) => {
   const body = JSON.stringify({
     includeTitle: true,
     includeSummary: true,
     includeRecommendations: true,
     includeDecisions: true,
-    meetingFromDate: now.toISOString(),
-    meetingToDate: nextMonth.toISOString(),
+    meetingFromDate: start.toISOString(),
+    meetingToDate: end.toISOString(),
   });
 
   const response = await cityCouncilXSRFPost({
-    url: 'https://secure.toronto.ca/council/api/multiple/agenda-items.json?pageNumber=0&pageSize=200&sortOrder=meetingDate%20asc,referenceSort',
+    url: `https://secure.toronto.ca/council/api/multiple/agenda-items.json?pageNumber=${page}&pageSize=200&sortOrder=meetingDate%20asc,referenceSort`,
     body,
   });
 
-  return ((await response.json()) as ApiResponse).Records;
+  return (await response.json()) as ApiResponse;
+};
+
+const deduplicateAgendaItems = (items: AgendaItem[]) => {
+  const seen = new Map<string, AgendaItem>();
+
+  items.forEach((item) => {
+    const key = `${item.reference}-${item.meetingId}`;
+    const prev = seen.get(key);
+    if (prev === undefined || item.agendaItemId >= prev.agendaItemId) {
+      seen.set(key, item);
+      if (prev !== undefined) {
+        console.log(
+          `ignoring ${prev.agendaItemId} in favor of ${item.agendaItemId}`,
+        );
+      }
+    } else {
+      console.log(
+        `ignoring ${item.agendaItemId} in favor of ${prev.agendaItemId}`,
+      );
+    }
+  });
+
+  return [...seen.values()];
+};
+
+export const fetchAgendaItems = async (opts: AgendaItemFetchOptions) => {
+  let pageNumber = 0;
+  let totalPages = 1;
+  const records: AgendaItem[] = [];
+
+  while (pageNumber < totalPages) {
+    const page = await fetchItemPage(pageNumber, opts);
```

Comment:
In future we could use a client wrapper to hold onto the CSRF to cut our requests in half
(The experimental version I had in the old repo took that approach so there's a [ref impl](https://github.com/civic-dashboard/civic-dashboard-web/pull/37/files#diff-c7ad86dda7418aa67c831f467da0b4861e4b08437cb1a9abecdc717eaf233968))

Reply:
agree! i was just too lazy to pass a client around and don't like having stateful global instances of things